listener/rhsm.go

```diff
@@ -0,0 +1,38 @@
+package listener
+
+import (
+	"app/base/models"
+	"app/base/utils"
+	"errors"
+
+	"gorm.io/gorm"
+)
+
+func getTemplate(db *gorm.DB, accountID int, environments []string) (*int64, error) {
+	var templateID *int64
+	if len(environments) == 0 {
+		// no environments, set template_id=NULL
+		return templateID, nil
+	}
+
+	// get template ids for given environments
+	var environmentTemplates []int64
+	err := db.Model(models.Template{}).
+		Where("rh_account_id = ? AND environment_id IN (?)", accountID, environments).
+		Select("id").
+		Scan(&environmentTemplates).Error
+	if err != nil {
+		return nil, err
+	}
+	if len(environmentTemplates) > 1 {
+		// TODO: should we support systems added to multiple templates?
```

Comment:
I'm going to check with Justin what to do with multiple envs...

Reply:
In case of multiple environments we agreed to just log a warning and assign system to the first one.